src/server/scripts/EasternKingdoms/TheStockade/instance_the_stockade.cpp

```diff
@@ -22,23 +22,53 @@ Without it, the party doing random dungeon won't get satchel of spoils and
 gets instead the deserter debuff.
 */
 
+
 #include "ScriptMgr.h"
 #include "InstanceScript.h"
+#include "the_stockade.h"
 
 class instance_the_stockade : public InstanceMapScript
 {
 public:
     instance_the_stockade() : InstanceMapScript("instance_the_stockade", 34) { }
 
+    struct instance_the_stockade_InstanceMapScript : public InstanceScript
+    {
+        instance_the_stockade_InstanceMapScript(Map* map) : InstanceScript(map)
+        {
+            SetHeaders(DataHeader);
+            SetBossNumber(EncounterCount);
+        }
+
+        void OnCreatureCreate(Creature* creature) override
+        {
+            switch (creature->GetEntry())
+            {
+            case NPC_RANDOLPH_MOLOCH:
+                RandolphMolochGuid = creature->GetGUID();
+                break;
+            case NPC_LORD_OVERHEAT:
+                LordOverheatGuid = creature->GetGUID();
+                break;
+            case NPC_HOGGER:
+                HoggerGUID = creature->GetGUID();
+                break;
+            default:
+                break;
+            }
+        }
+
+    protected:
+        ObjectGuid RandolphMolochGuid;
```

Comment:
Check other instancescript's usage of LoadObjectData to get rid of these variables